altapay.php

```diff
@@ -2295,7 +2295,7 @@ private function decodeHtmlEntitiesArrayValues($arr)
 
     public function getorderCartRule($orderID)
     {
-        $cartDiscount = Db::getInstance()->executeS('SELECT * FROM ' . _DB_PREFIX_ . 'order_cart_rule WHERE id_order = ' . $orderID);
+        $cartDiscount = Db::getInstance()->executeS('SELECT * FROM ' . _DB_PREFIX_ . 'order_cart_rule WHERE id_order = ' . (int) $orderID);
```

Comment:
Is the order ID always a numeric? Can't the merchant use letters?

Reply:
Yes, this column type is set as an int in the "order_cart_rule"  table and this is for internal use. The one which merchants can change is referenced id which is in string format.
<img width="926" alt="image" src="https://user-images.githubusercontent.com/15101585/207390477-a694335d-afb8-42bc-b907-88f0458713c8.png">